src/core/db/manager.py

```diff
@@ -33,7 +35,13 @@ def __exit__(self, type, value, traceback):
 
 class DataHubManager:
 
-    def __init__(self, user, repo_base=None, is_app=False):
+    def __init__(self, user=settings.ANONYMOUS_ROLE, repo_base=None,
+                 is_app=False):
+
+        # blank users are set to anonymous role
+        if user == '':
```

Comment:
`request.user.get_username()` == `''` when there is not a logged in user. This is triggered from the public view for unauthenticated users. I could move checks to view functions, to make them more explicit, or see whether I can override `get_username()`. What do you think?